package/MDAnalysis/analysis/dihedrals.py

```diff
@@ -453,6 +454,11 @@ def _single_frame(self):
                                     box=self.ag1.dimensions)
         phi_psi = [(phi, psi) for phi, psi in zip(phi_angles, psi_angles)]
         self.results.angles.append(phi_psi)
+    
+    def _setup_scheduler(self, scheduler, n_workers):
```

Comment:
I'm seeing a lot of repeats of this construct. Rather than doing this for all the default cases, would it not be better for this to be directly inherited as the default implementation of this method from the base class? It would avoid breaking a bunch of stuff downstream.

Reply:
This is now replaced by default `available_schedulers`, which in some cases are limited.